unified_planning/grpc/proto_writer.py

```diff
@@ -0,0 +1,534 @@
+        return unified_planning_pb2.Timing(
+            timepoint=self.convert(timing._timepoint),
+            delay=self._convert_to_real(timing.delay),
+        )
+
+    def _convert_to_real(self, element):
+        return self.convert(fractions.Fraction(element))
+
+    @handles(fractions.Fraction)
+    def _convert_fraction(self, fraction):
+        return unified_planning_pb2.Real(
+            numerator=fraction.numerator, denominator=fraction.denominator
+        )
+
+    @handles(unified_planning.model.timing.Interval)
+    def _convert_interval(self, interval):
+        return unified_planning_pb2.Interval(
+            is_left_open=interval.is_left_open(),
+            lower=self.convert(interval.lower()),
+            is_right_open=interval.is_right_open(),
+            upper=self.convert(interval.lower()),
+        )
+
+    @handles(unified_planning.model.TimeInterval)
+    def _convert_time_interval(self, interval):
+        return unified_planning_pb2.TimeInterval(
+            is_left_open=interval.is_left_open(),
+            lower=self.convert(interval.lower),
+            is_right_open=interval.is_right_open(),
+            upper=self.convert(interval.upper),
+        )
+
+    @handles(unified_planning.model.DurationInterval)
+    def _convert_duration_interval(self, interval):
+        return unified_planning_pb2.Duration(
+            controllable_in_bounds=unified_planning_pb2.Interval(
+                is_left_open=interval.is_left_open(),
+                lower=self.convert(interval.lower),
+                is_right_open=interval.is_right_open(),
+                upper=self.convert(interval.upper),
+            )
+        )
+
+    @handles(unified_planning.model.Problem)
+    def _convert_problem(self, problem):
+        goals = [unified_planning_pb2.Goal(goal=self.convert(g)) for g in problem.goals]
+        for (t, gs) in problem.timed_goals:
+            goals += [
+                unified_planning_pb2.Goal(goal=self.convert(g), timing=self.convert(t))
+                for g in gs
+            ]
+
+        return unified_planning_pb2.Problem(
+            domain_name=str(problem.name + "_domain"),
+            problem_name=problem.name,
+            types=[self.convert(t) for t in problem.user_types],
+            fluents=[self.convert(f, problem) for f in problem.fluents],
+            objects=[self.convert(o) for o in problem.all_objects],
+            actions=[self.convert(a) for a in problem.actions],
+            initial_state=[
+                unified_planning_pb2.Assignment(
+                    fluent=self.convert(x), value=self.convert(v)
+                )
+                for (x, v) in problem.initial_values.items()
+            ],
+            timed_effects=[self.convert(e) for e in problem.timed_effects],
+            goals=goals,
+            features=[map_feature(feature) for feature in problem.kind.features],
+            metrics=[self.convert(m) for m in problem.quality_metrics],
+        )
```

Comment:
Here we discard the `_initial_defaults` attribute

Reply:
No, it is included in the fluents, which is why the fluent conversion requires the problem as an additional argument.

Reply:
Not really. We have two different kind of defaults: one is the fluents default (`_fluents_defaults` attribute) and you are including it; the other one (`_initial_defaults` attribute) associate a default to the different `Type`, and can be set as a parameter in the `Problem` constructor (e.g. `problem = Problem('name', {BoolType(): False})`).

Reply:
I merged the PR #138

Reply:
Thanks, I rebased the PR so the fix is now in this branch.